torch/batching.py

```diff
@@ -0,0 +1,47 @@
+import torch
+import functools
+from torch import Tensor
+
+VMAP_LEVEL = 0
```

Comment:
Why does this need to be a global?

Reply:
Alternatively, it could be a thread_local. This value is keeping track of how many nested vmaps we are inside.